File: src/app/models/UsageRecord.js


// Define collection and schema for UsageRecord
export interface UsageRecord {
    periodStart:
	type : Date
    periodEnd:
	type : Date
    messagesSent:
	type : number
    dataVolumeMB:
	type : number
    Tenant:
	type : Schema.Types.ObjectId
    Device:
	type : Schema.Types.ObjectId
    ConnectivityPlan:
	type : Schema.Types.ObjectId
#
    collection: 'usageRecords'
}

File: iotonaspdotnet/Domain/UsageRecord.cs
using iotonaspdotnet.Contracts;

namespace iotonaspdotnet.Domain;

public class UsageRecord
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? UsagerecordId { get; set; } 
 public virtual DateOnly? PeriodStart { get; set; } 
 public virtual DateOnly? PeriodEnd { get; set; } 
 public virtual int? MessagesSent { get; set; } 
 public virtual int? DataVolumeMB { get; set; } 
public virtual Tenant? Tenant { get; set; } 
public virtual IoTDevice? Device { get; set; } 
public virtual ConnectivityPlan? ConnectivityPlan { get; set; } 

    public static UsageRecord FromRequest(UsageRecordRequest request) {
        return new UsageRecord {
            Id = request.Id,
            PeriodStart = request.PeriodStart,
            PeriodEnd = request.PeriodEnd,
            MessagesSent = request.MessagesSent,
            DataVolumeMB = request.DataVolumeMB,
        };
    }
}
